storage/node.go

```diff
@@ -233,15 +224,15 @@ func NewNodeIDForTreeCoords(depth int64, index int64, maxPathBits int) (NodeID,
 	// This node is effectively a prefix of the subtree underneath (for non-leaf
 	// depths), so we shift the index accordingly.
 	uidx := uint64(index) << uint(depth)
-	r := NewEmptyNodeID(maxPathBits)
+	r := NodeID{Path: make([]byte, maxPathBits/8)} // Note: maxPathBits % 8 == 0.
 	for i := len(r.Path) - 1; uidx > 0 && i >= 0; i-- {
 		r.Path[i] = byte(uidx & 0xff)
 		uidx >>= 8
 	}
 	// In the storage model nodes closer to the leaves have longer nodeIDs, so
 	// we "reverse" depth here:
 	r.PrefixLenBits = int(maxPathBits - int(depth))
-	return *r, nil
+	return r, nil
```

Comment:
For consistency all the New stuff should return pointers. Can be done later.

Reply:
Will do separately. Should they all return `NodeID` by value though? It's a lightweight type, and it's also meant to be immutable.

Reply:
Not sure how strong the Go convention is. The Effective Go examples return pointers.